core-features/tracing/index.test.js

```diff
@@ -0,0 +1,393 @@
+const { PrismaClient } = require('@prisma/client')
+const prisma = new PrismaClient()
+const axios = require('axios')
+
+function buildTree(tree, spans) {
+  const childrenSpans = spans.filter((span) => {
+    const [ref] = span.references
+    if (!ref) {
+      return false
+    }
+
+    return ref.refType === 'CHILD_OF' && ref.spanID === tree.span.spanID
+  })
+
+  if (childrenSpans.length) {
+    tree.children = childrenSpans.map((span) => buildTree({ span }, spans))
+  } else {
+    tree.children = []
+  }
+
+  return tree
+}
+
+describe('tracing', () => {
+  it('should request app and assert that traces are there', async () => {
+    const email = `${new Date().toISOString()}@email.com`
+    await prisma.user.create({
+      data: {
+        email,
+      },
+    })
+
+    const appResponse = await axios({
+      method: 'GET',
+      url: `http://localhost:4000/users`,
+    })
+    expect(appResponse.status).toEqual(200)
+
+    const jeagerResponse = await axios({
+      method: 'GET',
+      url: 'http://localhost:16686/api/traces?service=ecosystem-tests-core-features-tracing',
+    })
+    expect(jeagerResponse.status).toEqual(200)
+
+    const [trace] = jeagerResponse.data.data
+    const [rootSpan, ...restOfSpans] = trace.spans
+    const traceTree = buildTree({ span: rootSpan }, restOfSpans)
```

Comment:
I think you might need to order the JSON first to have a stable snapshot below? or find another approach

Reply:
The functional tests now implement a more advanced `buildTree`, although that looks a bit dangerous that it might actually delete useful information: https://github.com/prisma/prisma/blob/f46bca1625018ebf94ee1b3242cc40c3539c49ea/packages/client/tests/functional/tracing/tests.ts#L25-L74